tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs

```diff
@@ -263,7 +269,8 @@ protected virtual void ModifyOperationIfNeeded(BsonDocument operation)
         protected virtual void RunTest(BsonDocument shared, BsonDocument test, EventCapturer eventCapturer)
         {
             using (var client = CreateDisposableClient(test, eventCapturer))
-            {
+            using (ConfigureTestClientRelatedDisposable())
```

Comment:
Why not just override `ServerDiscoveryAndMonitoringIntegrationTestRunner.RunTest`?

Reply:
all changes here are reverted